File: ProblemSolve/1359.cs
/***********************************************
 * Count All Valid Pickup and Delivery Options *
 ***********************************************/

public class Solution {
    public int CountOrders(int n) {
        ulong mod = 1000000007;
        ulong ans = 1;
        
        for(uint i=1; i<=n; ++i){
            ans = ans * i;            
            ans %= mod;            
            ans = ans * (2*i - 1);
            
            ans %= mod;            
        }
        
        return (int)ans;
    }
}

File: ProblemSolve/1359.cpp
/***********************************************
 * Count All Valid Pickup and Delivery Options *
 ***********************************************/

#define mod 1000000007

class Solution {
public:
    int countOrders(int n) {
        long long ans = 1;
        
        for(int i=1; i<=n; ++i){
            ans *= i;
            ans %= mod;
            
            ans *= (2*i - 1);
            ans %= mod;
        }
        
        return ans;
    }
};